CONTRIBUTING.rst

```diff
@@ -0,0 +1,194 @@
+would prefer ``tox`` to use ``conda``, run:
+
+.. code-block:: console
+
+   $ pip install tox-conda
+
+Running Part of the Tests
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+During development, you likely would like to run only part of the tests to save time.
+
+To run all static tests, run:
+
+.. code-block:: console
+
+   $ tox -e lint
+
+To run all runtime tests on the Python version in the development environment, run:
+
+.. code-block:: console
+
+   $ tox -e py
+
+To run only a specific runtime test, run:
+
+.. code-block:: console
+
+   $ pytest tests -vk [test_name]  # e.g., pytest tests -vk test_default_data_dir
+
+Read `pytest command line document <https://docs.pytest.org/en/stable/usage.html>`__ for its more advanced usage.
+
+To run document generation tests, run:
+
+.. code-block:: console
+
+   $ tox -e docs
+
+Where to Expose a Symbol (Function, Class, etc.)?
+-------------------------------------------------
+
+Generally speaking:
+
+- If a symbol is likely used by a casual user regularly, it should be exposed in :file:`pydax/__init__.py`. This gives
+  casual users the cleanest and the most direct access.
+- If a symbol is used only by a power user, but is unlikely used by a casual user regularly, it should be exposed in a
+  file that does not start with an underscore, such as :file:`pydax/schema.py`; or in the :file:`__init__.py` file in a
+  subdirectory that does not start with an underscore, such as :file:`pydax/loaders/__init__.py`. The rationale is that
+  the amount of such symbols is usually large and if we expose them at the root level of the package, it would be messy
+  and likely confuse casual users.
+- If a symbol is solely used for internal purpose, it should be exposed only in files starting with a single underscore,
+  such as :file:`pydax/_dataset.py`.
+
+Please keep in mind that the criteria above are not meant to be rigid: They should be applied flexibly in light of
+factors such as where existing symbols are placed and other potentially important considerations (if any).
+
+## Where to Import a Symbol?
+
+When referencing a symbol that is exposed to a user, in general, we prefer importing the symbol from where the package
+publicly exposes it over importing from where the source code of the symbol is defined, e.g., use ``from .schema import
+SchemaDict`` rather than ``from ._schema import SchemaDict``. This way we have more code paths that would go through
+what the user would actually experience and hopefully would give us more chances to discover bugs.
+
+## Continuous Integration (CI)
+
+We prefer keeping CI configuration files, namely :file:`.travis.yml` and :file:`.appveyor.yml` simple and unscrambled.
+Normally, only test environment, such as Python version, OS and tox environmental variables, or anything that is
+specific to the CI system, such as failure notification. Complicated test dependencies and other test dealings should go
+to :file:`tox.ini` and their respective test files in :file:`tests/`.
+
+## Docs
```

Comment:
Missed the subheadings here:
`## Where to Import a Symbol?`, `## Continuous Integration (CI)`, `## Docs`